src/lib/stores/Quaff.ts

```diff
@@ -10,30 +10,37 @@ function quaff() {
     //TODO? iconSet: {},
   });
 
+  type Mode = "light" | "dark";
+
   const toggleDarkMode = () => {
-    return update((q) => {
-      q.dark = !q.dark;
+    return update(($quaff) => {
+      $quaff.dark = !$quaff.dark;
+      const mode: { from: Mode; to: Mode } = $quaff.dark
+        ? {
+            from: "light",
+            to: "dark",
+          }
+        : {
+            from: "dark",
+            to: "light",
+          };
 
       let body = document.querySelector("body");
-
-      if (q.dark === true) {
-        body && body.classList.replace("light", "dark");
-      } else {
-        body && body.classList.replace("dark", "light");
+      if (body) {
+        body.classList.replace(`body--${mode.from}`, `body--${mode.to}`);
```

Comment:
While I like the conciseness of it, the downside is you have to set a `body--light` class initially in your layout. This makes it somewhat fragile from an engineering standpoint. We should think if we can find another way, while keeping the code short. Also, I think in case the `q.dark` mismatches the class on the layout, it will be even more problematic and stop working completely.

Reply:
Don't worry, I'm caching the mode inside cookies to be able to read it in server side.
If there's no cookie, the mode is already automatically set to "light" and a cookie is created on server side. So when this code runs, there's always a cookie